src/Microsoft.AspNetCore.Blazor.Build/Core/BootJsonWriter.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.JSInterop;
+using Mono.Cecil;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.AspNetCore.Blazor.Build
+{
+    internal class BootJsonWriter
+    {
+        public static void WriteFile(
+            string assemblyPath,
+            string[] assemblyReferences,
+            string[] embeddedResourcesSources,
+            bool linkerEnabled,
+            string outputPath)
+        {
+            var embeddedContent = EmbeddedResourcesProcessor.ExtractEmbeddedResources(
+                embeddedResourcesSources, Path.GetDirectoryName(outputPath));
+            var bootJsonText = GetBootJsonContent(
+                Path.GetFileName(assemblyPath),
+                GetAssemblyEntryPoint(assemblyPath),
+                assemblyReferences,
+                embeddedContent,
+                linkerEnabled);
+            var normalizedOutputPath = Normalize(outputPath);
+            Console.WriteLine("Writing boot data to: " + normalizedOutputPath);
+            File.WriteAllText(normalizedOutputPath, bootJsonText);
+        }
+
+        public static string GetBootJsonContent(string assemblyFileName, string entryPoint, string[] assemblyReferences, IEnumerable<EmbeddedResourceInfo> embeddedContent, bool linkerEnabled)
+        {
+            var data = new BootJsonData(
+                assemblyFileName,
+                entryPoint,
+                assemblyReferences,
+                embeddedContent,
+                linkerEnabled);
+            return Json.Serialize(data);
+        }
+
+        private static string GetAssemblyEntryPoint(string assemblyPath)
+        {
+            using (var assemblyDefinition = AssemblyDefinition.ReadAssembly(assemblyPath))
+            {
+                var entryPoint = assemblyDefinition.EntryPoint;
+                if (entryPoint == null)
+                {
+                    throw new ArgumentException($"The assembly at {assemblyPath} has no specified entry point.");
+                }
+
+                return $"{entryPoint.DeclaringType.FullName}::{entryPoint.Name}";
+            }
+        }
+
+        private static string Normalize(string outputPath) =>
+            Path.Combine(Path.GetDirectoryName(outputPath), Path.GetFileName(outputPath).ToLowerInvariant());
```

Comment:
Not sure if this is part of an earlier version, but how does this work on a case-sensitive filesystem?

Reply:
That's a very good point! It looks like it *wouldn't* work on a case-sensitive filename.

I retained this logic from the earlier `IndexHtmlWriter`. I could try to ask why the `Normalize` step was added there (wasn't me) but TBH it looks like it might just have been for aesthetic reasons. Without it, the build works fine but logs lines like this:

```
Writing boot data to: C:\Users\username\Some\Dir\StandaloneApp\obj\Debug\netstandard2.0\blazor/blazor.boot.json
```

... so maybe it was just to deal with the weirdness of the slashes.

I've now replaced the `Normalize` method with just `Path.GetFullPath(outputPath)`. This corrects the slashes (and would resolve relative paths or directory traversal segments) without interfering with anything else.